File: mobile/lib/models/search_result.dart
import 'video_model.dart';

/// 搜索结果
class SearchResult {
  final List<VideoModel> videos;
  final int page;
  final int pageSize;
  final int numResults;
  final int numPages;

  const SearchResult({
    required this.videos,
    required this.page,
    required this.pageSize,
    required this.numResults,
    required this.numPages,
  });

  bool get hasMore => page < numPages;
}

File: windows/Models/SearchResult.cs
using System.Collections.Generic;

namespace bilibili_music_player_windows.Models
{
    public sealed class SearchResult
    {
        public SearchResult(IReadOnlyList<VideoPreviewItem> videos, int page, int pageSize, int numResults, int numPages)
        {
            Videos = videos;
            Page = page;
            PageSize = pageSize;
            NumResults = numResults;
            NumPages = numPages;
        }

        public IReadOnlyList<VideoPreviewItem> Videos { get; }
        public int Page { get; }
        public int PageSize { get; }
        public int NumResults { get; }
        public int NumPages { get; }
    }
}
